src/blueapi/client/numtracker.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from collections.abc import Mapping
+from pathlib import Path
+from textwrap import dedent
+
+import requests
+from pydantic import Field
+
+from blueapi.utils import BlueapiBaseModel
+
+
+class NumtrackerNewScanVisit(BlueapiBaseModel):
+    instrument: str
+    instrument_session: str = Field(alias="instrumentSession")
+    path: Path
+
+
+class NumtrackerNewScanScan(BlueapiBaseModel):
+    scan_file: str = Field(alias="scanFile")
+    scan_number: int = Field(alias="scanNumber")
+    directory: NumtrackerNewScanVisit
+
+
+class NumtrackerNewScan(BlueapiBaseModel):
+    scan: NumtrackerNewScanScan
+
+
+class NumtrackerClient:
+    def __init__(
+        self,
+        url: str,
+        headers: Mapping[str, str],
+    ) -> None:
+        self._url = url
+        self._headers = headers
+
+    def set_headers(self, pass_through_headers: Mapping[str, str]) -> None:
+        self._headers = pass_through_headers
+
+    # TODO: Could make this async, but since it's called from RE.scan_id_source, we
+    # would need to change the RE to accept an async function in the scan_id_source
+    # hook. It's a 1-line change but would need to be reviewed etc.
```

Comment:
Should: Replace TODO with an issue, maybe on upstream bluesky?

Reply:
I made an issue but wasn't entirely sure of the details on what to write. https://github.com/bluesky/bluesky/issues/1895